concent_api/core/views.py

```diff
@@ -43,7 +43,10 @@ def send(request, client_message):
     elif isinstance(client_message, message.concents.ForceSubtaskResults) and client_message.ack_report_computed_task is not None:
         return handle_send_force_subtask_results(request, client_message)
 
-    elif isinstance(client_message, message.concents.ForceSubtaskResultsResponse):
+    elif (
+        isinstance(client_message, message.concents.ForceSubtaskResultsResponse) and
+        (client_message.subtask_results_accepted is not None or client_message.subtask_results_rejected is not None)
```

Comment:
Shouldn't you also use `isinstance()` on `subtask_results_accepted` and `subtask_results_rejected`? What happens if they're different types of messages? Or not messages at all?

Reply:
@cameel As I thought, it will then fail during load(). Here is the output from console:

```
TypeError: Invalid nested message type <class 'int'> should be <class 'golem_messages.message.tasks.SubtaskResultsRejected'>
FieldError:  [subtask_results_rejected:1]
TypeError: Invalid nested message type <class 'golem_messages.message.concents.ServiceRefused'> should be <class 'golem_messages.message.tasks.SubtaskResultsRejected'>
FieldError:  [subtask_results_rejected:ServiceRefused(header=MessageHeader(type_=4000, timestamp=1518447887, encrypted=False), sig=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', slots=[['subtask_id', None], ['task_to_compute', None], ['reason', None]])]

```

Reply:
OK. So we need to detect it before it happens.

As a general rule, the client should not be able to cause HTTP 500 by simply sending us invalid data.

Reply:
We are catching this in `api_views.py`.